diff --git a/menu_block.module b/menu_block.module
index ae1c4c4..2ad8472 100644
--- a/menu_block.module
+++ b/menu_block.module
@@ -4,8 +4,11 @@
  * Provides configurable blocks of menu items.
  */
 
-// A core bug in Drupal 7.0 requires this fix.
-module_load_include('inc', 'menu_block', 'custom_menu_active_trail');
+// A core bug fixed in Drupal 7.12.
+// @see http://drupal.org/node/942782
+if (!defined('MENU_PREFERRED_LINK')) {
+  module_load_include('inc', 'menu_block', 'custom_menu_active_trail');
+}
 
 /**
  * Denotes that the tree should use the menu picked by the curent page.
@@ -309,7 +312,9 @@ function menu_tree_build($config) {
   menu_block_set_title(t($menu_names[$config['menu_name']]));
 
   // @TODO: Remove work-around for core bug #942782.
-  __menu_block_set_menu_trail($config['menu_name']);
+  if (!defined('MENU_PREFERRED_LINK')) {
+    __menu_block_set_menu_trail($config['menu_name']);
+  }
 
   if ($config['expanded'] || $config['parent_mlid']) {
     // Get the full, un-pruned tree.
@@ -323,7 +328,9 @@ function menu_tree_build($config) {
   }
 
   // @TODO: Remove work-around for core bug #942782.
-  __menu_block_set_menu_trail();
+  if (!defined('MENU_PREFERRED_LINK')) {
+    __menu_block_set_menu_trail();
+  }
 
   // Allow alteration of the tree and config before we begin operations on it.
   drupal_alter('menu_block_tree', $tree, $config);
