diff --git a/better_exposed_filters.theme b/better_exposed_filters.theme
index 18ff6e9..0a79b72 100644
--- a/better_exposed_filters.theme
+++ b/better_exposed_filters.theme
@@ -80,6 +80,11 @@ function theme_select_as_checkboxes($vars) {
     return theme('select_as_tree', array('element' => $element));
   }
 
+  // if the multiple attribute is set it must be removed because it's not a valid attribute for the checkbox input element
+  if (isset($element['#attributes']['multiple'])) {
+    unset($element['#attributes']['multiple']);
+  }
+
   // The selected keys from #options.
   $selected_options = empty($element['#value']) ? $element['#default_value'] : $element['#value'];
 
@@ -145,6 +150,10 @@ function theme_select_as_checkboxes($vars) {
     $attributes['class'] = array_merge($element['#attributes']['class'], $attributes['class']);
   }
 
+  // Remove the name attribute because it is not valid for the div element
+  if (isset($element['#attributes']['name'])) {
+    unset($element['#attributes']['name']);
+  }
   return '<div' . drupal_attributes($attributes) . ">$description$output</div>";
 }
 
